src/vs/platform/terminal/common/capabilities/shellEnvDetectionCapability.ts

```diff
@@ -36,4 +37,33 @@ export class ShellEnvDetectionCapability extends Disposable implements IShellEnv
 		// Convert to event and fire event
 		this._onDidChangeEnv.fire(this._env);
 	}
+
+	startEnvironmentSingleVar(isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		this._env.clear();
+		this._pendingEnv = new Map();
+	}
+	setEnvironmentSingleVar(key: string, value: string | undefined, isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		if (this._pendingEnv && key !== undefined && value !== undefined) {
+			this._pendingEnv.set(key, value);
+		}
+	}
+	endEnvironmentSingleVar(isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		if (this._pendingEnv) {
+			this._env.clear();
+			this._pendingEnv.forEach((value, key) => {
+				this._env.set(key, value);
+			});
+		}
+		this._pendingEnv = undefined;
+		this._onDidChangeEnv.fire(this._env);
```

Comment:
It simplifies things externally if we just replace `_env` like this:

```suggestion
		if (!this._pendingEnv) {
			return;
		}
		this._env = this._pendingEnv;
		this._pendingEnv = undefined;
		this._onDidChangeEnv.fire(this._env);
```

You'll need to remove `readonly` from `_env` or this will break compile.